src/control/ToolHandler.cpp

```diff
@@ -250,37 +256,38 @@ void ToolHandler::setLineStyle(const LineStyle& style) {
  * 			and therefore should not be applied to a selection
  */
 void ToolHandler::setColor(int color, bool userSelection) {
-    this->current->setColor(color);
+    this->toolbarSelectedTool->setColor(color);
+    this->buttonSelectedTool->setColor(color);
```

Comment:
I think this should only change the current selected tools color. Since it might be a bit confusing, having the high lighter on the pen button and changing with it the color of the pen.

Reply:
Hmm, good point. Since I'm only using the pen button as eraser I didn't think of this. I guess best would be:

```cpp
if (this->buttonSelectedTool->tool == TOOL_ERASER)
   this->toolbarSelectedTool->setColor(color);
```

What do you think?

Reply:
```suggestion
    if (this->buttonSelectedTool->tool == TOOL_ERASER) {
        this->toolbarSelectedTool->setColor(color);
    } else {
        this->currentTool->setColor(color);
    }
```

Reply:
Nice one! Yeah that would be ideal. I will only be able to wrap it up next week. I would include this change and then rebase on master.

Reply:
It Turned out that it was actually more beneficial to do the following, as setting the color for the current Tool even if it's a Eraser (selected by the button) ensures that in the toolbar the correct color is highlighted.

```cpp
    if (this->buttonSelectedTool->tool == TOOL_ERASER) {
        this->toolbarSelectedTool->setColor(color);
    }
    this->currentTool->setColor(color); 
```